src/pymmcore_widgets/hcs/_plate_graphics_scene.py

```diff
@@ -0,0 +1,158 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, cast
+
+import numpy as np
+from qtpy.QtCore import QRect, QRectF, Qt, Signal
+from qtpy.QtGui import QBrush, QColor, QTransform
+from qtpy.QtWidgets import (
+    QGraphicsItem,
+    QGraphicsScene,
+    QGraphicsSceneMouseEvent,
+    QRubberBand,
+    QWidget,
+)
+
+from ._graphics_items import GREEN
+
+if TYPE_CHECKING:
+    from ._graphics_items import Well, _WellGraphicsItem
+
+SELECTED_COLOR = QBrush(QColor(GREEN))
+UNSELECTED_COLOR = QBrush(Qt.GlobalColor.lightGray)
+
+
+class _PlateGraphicsScene(QGraphicsScene):
+    """Custom QGraphicsScene to control the plate/well selection.
+
+    To get the list of selected well info, use the `value` method
+    that returns a list of snake-row-wise ordered tuples (name, row, column).
+    """
+
+    valueChanged = Signal()
+
+    def __init__(self, parent: QWidget | None = None) -> None:
+        super().__init__(parent)
+
+        self.well_pos = 0
+        self.new_well_pos = 0
+
+        self._selected_wells: list[QGraphicsItem] = []
+
+    def mousePressEvent(self, event: QGraphicsSceneMouseEvent) -> None:
+        # origin point of the SCREEN
+        self.origin_point = event.screenPos()
+        # rubber band to show the selection
+        self.rubber_band = QRubberBand(QRubberBand.Shape.Rectangle)
+        # origin point of the SCENE
+        self.scene_origin_point = event.scenePos()
+
+        # get the selected items
+        self._selected_wells = [item for item in self.items() if item.isSelected()]
+
+        # set the color of the selected wells to SELECTED_COLOR if they are within the
+        # selection
+        for item in self._selected_wells:
+            item = cast("_WellGraphicsItem", item)
+            item.brush = SELECTED_COLOR
+
+        # if there is an item where the mouse is pressed and it is selected, deselect,
+        # otherwise select it.
+        if well := self.itemAt(self.scene_origin_point, QTransform()):
+            well = cast("_WellGraphicsItem", well)
+            if well.isSelected():
+                well.brush = UNSELECTED_COLOR
+                well.setSelected(False)
+            else:
+                well.brush = SELECTED_COLOR
+                well.setSelected(True)
+        self.valueChanged.emit()
+
+    def mouseMoveEvent(self, event: QGraphicsSceneMouseEvent) -> None:
+        # update the rubber band geometry using the SCREEN origin point and the current
+        self.rubber_band.setGeometry(QRect(self.origin_point, event.screenPos()))
```

Comment:
do you know if it's literally impossible for mouseMoveEvent to ever be called before a mousePressEvent has first been called on this Scene?  if not, both .rubber_band and .scene_origin_point will be undefined here

Reply:
I did some test and it seems that it won't be called unless a mousePressEvent  happens before.